validator_derive/src/quoting.rs

```diff
@@ -223,6 +223,66 @@ pub fn quote_length_validation(
     unreachable!()
 }
 
+pub fn quote_length_utf16_validation(
+    field_quoter: &FieldQuoter,
+    validation: &FieldValidation,
+) -> proc_macro2::TokenStream {
+    let field_name = &field_quoter.name;
+    let validator_param = field_quoter.quote_validator_param();
+
+    if let Validator::LengthUTF16 { min, max, equal } = &validation.validator {
```

Comment:
For consistency with Rust code style, you might want to use `Utf` in identifiers.

E.g. https://doc.rust-lang.org/std/str/struct.EncodeUtf16.html